Felienne/hedy
# Refactor stats pages and Add log search to the class stats page #1242

**Description**

- Refactor both stats pages so that they share code (through using jinja macros)
- Add shared legends for exceptions and per week charts on the admin stats page.
- Add log search for teachers in the class stats page

**Fix for**

#1242 

**How to test**

_Local test of the admin stats page_
Log in as an admin and open /stats page.
- Check the shared legends of the exceptions and the per-week charts. Do they look ok?
- Do the rest of the graphs look ok?
- Do the accordion tabs open and close correctly?
- The start and end date of the log search form should be pre-filled with today's date and they should both contain time.

_Local test of the class stats page_
Log in as either the class teacher or an admin and go to a /stats/class/$class_id page.
- Check the graphs and the shared legends
- Do they look as expected?
- Do the accordion tabs open and close correctly?
- The start and end date of the log search form should be pre-filled with today's date and they should both contain time.

Unfortunately, logs can only be tested on Alpha.
_Alpha test of the admin stats page_
Log in as an admin and go to the /stats page.
- You should be able to search for any username, e.g. hedy, hedy% or %hedy%
- In the results table, the copy code button should successfully copy code to the clipboard

_Alpha test of the class stats page_
- Log in as a teacher and to the /stats/class/$class-id page. In the log search form you should be able to select from a dropdown only the students of this class.
- In the results table, the copy code button should successfully copy code to the clipboard

**Checklist**

If you're unsure about any of these, don't hesitate to ask. We're here to help!
